alchemist/alchemist-incarnation-biochemistry/src/main/java/it/unibo/alchemist/model/implementations/conditions/BiomolPresentInNeighbor.java

```diff
@@ -61,23 +62,26 @@ public boolean isValid() {
             return false;
         } else {
             final Neighborhood<Double> neighborhood = environment.getNeighborhood(getNode());
-            return neigh.entrySet().stream().allMatch(n -> neighborhood.contains(n.getKey()) && n.getKey().getConcentration(mol) >=  conc);
+            return neigh.entrySet().stream()
+                    .filter(n -> n.getKey() instanceof CellNode)
+                    .allMatch(n -> neighborhood.contains(n.getKey()) 
+                            && n.getKey().getConcentration(mol) >=  conc);
         }
     }
 
     @Override
     public BiomolPresentInNeighbor cloneOnNewNode(final Node<Double> n) {
-        return new BiomolPresentInNeighbor(mol, conc, n, environment);
+        return new BiomolPresentInNeighbor(environment, n, mol, conc);
     }
 
     @Override
     public Map<Node<Double>, Double> getValidNeighbors(final Collection<? extends Node<Double>> neighborhood) {
         propensity = 0;
         neigh = neighborhood.stream()
-                    .filter(n -> n.getConcentration(mol) >= conc)
-                    .collect(Collectors.<Node<Double>, Node<Double>, Double>toMap(
-                                          n -> n,
-                                          n -> CombinatoricsUtils.binomialCoefficientDouble(n.getConcentration(mol).intValue(), conc.intValue())));
+                .filter(n -> n instanceof CellNode && n.getConcentration(mol) >= conc)
+                .collect(Collectors.<Node<Double>, Node<Double>, Double>toMap(
```

Comment:
That's because the return type of this method is forced to be Map<Node<Double>, Double> by AbstractNeighborCondition<T>. This method overrides **Map<Node<T>, T> getValidNeighbors**, so i can't return a Map<CellNode, Double>.